modules/run-vault/run-vault

```diff
@@ -137,11 +143,30 @@ function generate_vault_config {
   local readonly enable_s3_backend="$8"
   local readonly s3_bucket="$9"
   local readonly s3_bucket_region="${10}"
+  local readonly enable_auto_unseal="${11}"
+  local readonly auto_unseal_kms_key="${12}"
+  local readonly auto_unseal_kms_key_region="${13}"
   local readonly config_path="$config_dir/$VAULT_CONFIG_FILE"
 
   local instance_ip_address
   instance_ip_address=$(get_instance_ip_address)
 
+  local auto_unseal_config=""
+  if [[ "$enable_auto_unseal" == "true" ]]; then
+    auto_unseal_config=$(cat <<EOF
+seal "awskms" {
+  kms_key_id = "$auto_unseal_kms_key"
```

Comment:
Ah, OK, the Vault API uses `kms_key_id` as the param name, so we should probably follow their lead.

Reply:
I actually disagree. Reading through this code, I still don't know if Vault is expecting the key ARN or name here. I prefer to make our interfaces clear so that users know what to enter, sometimes making our var names clearer than the features they wrap.

Reply:
See above. CMK ID is a well-defined term.